stan/math/rev/core/vari.hpp

```diff
@@ -189,10 +180,184 @@ using vari = vari_value<double>;
 template <typename T, typename = void>
 class vari_view;
 
+/**
+ * This struct is follows the CRTP for methods common to `vari_view<>` and
+ * `vari_value<Matrix>`.
+ */
 template <typename T>
-class vari_view<T, require_all_t<bool_constant<!is_plain_type<T>::value>,
-                                 is_eigen_dense_base<T>>>
-    final : public vari_base {
+struct vari_view_crtp {
```

Comment:
@t4c1 if we can define these for `matrix_cl` can just plug this in! Then I think all the `rvalue()` subsetting functions I'm doing should also work for `matrix_cl` as well!

Reply:
That should be possible.